### PR TITLE
Internal 723 teardown plugin

### PR DESCRIPTION
**Pull request by jgrillo-grapl**
### Which issue does this PR correspond to?
https://app.zenhub.com/workspaces/grapl-6036cbd36bacff000ef314f2/issues/grapl-security/issue-tracker/723

### What changes does this PR make to Grapl? Why?
This PR adds a `tear_down_plugin` RPC endpoint which stops the Nomad job and marks the plugin as `deployed = false` in plugin-registry.

### How were these changes tested?
Automated tests.

### CODE REVIEW

File: src/rust/plugin-registry/src/nomad/client.rs [hunk 1]
```diff
@@ -137,6 +139,24 @@ impl NomadClient {
...
+    pub async fn delete_job(
+        &self,
+        job_name: String,
+        namespace: Option<String>,
+    ) -> Result<models::JobDeregisterResponse, NomadClientError> {
+        jobs_api::delete_job(
+            &self.internal_config,
+            jobs_api::DeleteJobParams {
+                namespace: namespace.clone(),
+                job_name: job_name.to_owned(),
```
Comment (colin-grapl) [inline src/rust/plugin-registry/src/nomad/client.rs:153]: I think both `.clone()` and `.to_owned()` are unnecessary here

**Review by colin-grapl — APPROVED** [review]

File: src/rust/plugin-registry/src/nomad/client.rs (hunk 1 above)
Comment (jgrillo-grapl) [reply src/rust/plugin-registry/src/nomad/client.rs]: fixed

File: src/rust/plugin-registry/src/db/client.rs [hunk 2]
```diff
@@ -112,7 +112,8 @@ impl PluginRegistryDbClient {
                 id,
                 plugin_id,
                 deploy_time,
-                status AS "status: PluginDeploymentStatus"
+                status AS "status: PluginDeploymentStatus",
+                deployed
```
Comment (wittekm) [inline src/rust/plugin-registry/src/db/client.rs:116]: PluginDeploymentStatus not containing 'is_deployed' is not ideal, but whatever it's fine for now :)

File: src/rust/plugin-registry/src/nomad/client.rs [hunk 3]
```diff
@@ -137,6 +139,24 @@ impl NomadClient {
         .await
         .map_err(NomadClientError::from)
     }
+
+    #[tracing::instrument(skip(self, job_name, namespace), err)]
+    pub async fn delete_job(
```
Comment (wittekm) [inline src/rust/plugin-registry/src/nomad/client.rs:144]: congrats you're now the second person to touch the nomad-rs stuff. it's a lil wonky but not too awful

File: src/rust/plugin-registry/src/server/deploy_plugin.rs [hunk 4]
```diff
@@ -167,6 +167,30 @@ pub async fn deploy_plugin(
     Ok(())
 }
 
+#[tracing::instrument(skip(client, db_client, plugin), err)]
+pub async fn teardown_plugin(
```
Comment (wittekm) [inline src/rust/plugin-registry/src/server/deploy_plugin.rs:171]: Unspecified behavior: what do we return if the plugin is already torn down?

**Review by wittekm — CHANGES_REQUESTED** [review]

File: src/rust/rust-proto/src/graplinc/grapl/api/plugin_registry/v1beta1.rs [hunk 5]
```diff
@@ -364,6 +367,236 @@ impl ProtobufSerializable for DeployPluginResponse {
     type ProtobufMessage = proto::DeployPluginResponse;
 }
 
+//
+// PluginDeployment
+//
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum PluginDeploymentStatus {
+    Unspecified,
```
Comment (wimax-grapl) [inline src/rust/rust-proto/src/graplinc/grapl/api/plugin_registry/v1beta1.rs:376]: i'd say that we should probably just follow the example of other rust-proto enums where Unspecified is never really expected to be handled at runtime? but i'm not committed to that

File: src/rust/plugin-registry/src/server/deploy_plugin.rs (hunk 4 above)
Comment (jgrillo-grapl) [reply src/rust/plugin-registry/src/server/deploy_plugin.rs:171]: fixed

File: src/rust/plugin-registry/src/db/client.rs [hunk 6]
```diff
@@ -196,4 +197,25 @@ impl PluginRegistryDbClient {
...
+            r"
+            UPDATE plugin_deployment
+                SET deployed = false
+            WHERE
+                id = $1
+                AND plugin_id = $2::uuid;
+            ",
+            plugin_deployment_row.id,
+            plugin_deployment_row.plugin_id,
+        )
```
[the remark this answers is not in the document]
Comment (jgrillo-grapl) [reply src/rust/plugin-registry/src/db/client.rs]: fixed

**Review by wimax-grapl — APPROVED** [review]

**Pull request merged**